src/serial.rs

```diff
@@ -96,11 +97,32 @@ pub mod config {
         STOP1P5,
     }
 
+    pub enum BitOrder {
+        Lsb,
+        Msb
+    }
+
+    pub enum ClockPhase {
+        First,
+        Second
+    }
+
+    pub enum ClockPolarity {
+        High,
+        Low
+    }
+
     pub struct Config {
         pub baudrate: Hertz,
         pub wordlength: WordLength,
         pub parity: Parity,
         pub stopbits: StopBits,
+        pub clock_phase: ClockPhase,
+        pub bit_order: BitOrder,
+        pub clock_polarity: ClockPolarity,
+
+        /// TODO: should we rename this?
+        pub clock_last_data_pulse: bool
```

Comment:
The manual refers to this as `Last Bit CLock pulse` (LBCL)- can we rename to that and make it an enum like the others?